### ISSUE TITLE
Improve menu hover/focus functionality

### ISSUE THREAD

**Issue by allisonacs** [zendeskgarden/react-components#256]
## Expectations

I expect menu hover/focus styling to track with my mouse. If I hover over items, I expect them to get focus styling. Once my mouse is no longer focusing on that item, I expect it to lose focus styling. I expect the most recently used input type to take control of the menu (e.g. if I am focusing, and I start using the keyboard, I expect the keyboard to take over from where the mouse was. If the mouse exits the menu area, but the menu is still open, and I start using the keyboard, I expect the highlighting to start from the top of the menu).

Basically I expect the same behavior present in native menus.

## Reality

Mouse can leave the menu area, but items are still highlighted. This is also true if you focus a disabled item; the hover state remains on the last focused item. 

## Steps to Reproduce

  1. Use a Garden menu, observe hover/focus treatments

###### Fine Print

* Component: Menus
* Browsers: all

**Comment by jzempel** [comment]
Addressed by https://github.com/zendeskgarden/react-components/pull/168
